src/DonationForms/V2/DataTransferObjects/DonationFormQueryData.php

```diff
@@ -127,4 +152,83 @@ public function getDonationFormLevels($object): array
                 return [];
         }
     }
+
+
+    /**
+     * @unreleased
+     */
+    private function getGoalSettings(object $queryObject): GoalSettings
+    {
+        $currency = give_get_option('currency', 'USD');
+        $formSettings = $queryObject->{DonationFormMetaKeys::SETTINGS()->getKeyAsCamelCase()};
+
+        // v3 form
+        if ($formSettings) {
+            $settings = FormSettings::fromjson($formSettings);
+            // uses campaign goal settings
+            if ($settings->goalSource->isCampaign()) {
+                $campaign = Campaign::findByFormId($queryObject->id);
+                $this->campaignId = $campaign->id;
+
+                $goalType = $this->convertGoalType(
+                    $campaign->goalType->getValue(),
+                    (bool)$queryObject->recurringGoalFormat
+                );
+
+                return GoalSettings::fromArray([
+                    'goalSource' => $settings->goalSource->getValue(),
+                    'enableDonationGoal' => $settings->enableDonationGoal,
+                    'goalType' => $goalType,
+                    'goalAmount' => $goalType->isOneOf(GoalType::AMOUNT(), GoalType::AMOUNT_FROM_SUBSCRIPTIONS())
+                        ? Money::fromDecimal($campaign->goal, $currency)->formatToDecimal()
+                        : $campaign->goal,
+                ]);
```

Comment:
Something's not right about this for v3 forms as I change goal types on campaign.  I think this part might be the culprit as we should be able to use form settings directly instead of trying to convert them. 

```suggestion
                return GoalSettings::fromArray([
                    'goalSource' => $settings->goalSource->getValue(),
                    'enableDonationGoal' => $settings->enableDonationGoal,
                    'goalType' => $settings->goalType,
                    'goalAmount' => $settings->goalAmount,
                ]);
```

Reply:
Ah yeah, you got it working in the DonationFormGoalData somehow.  If you set a form goal to inherit campaign and change the campaign settings, the form list should match here:

<img width="1294" alt="Screenshot 2025-05-13 at 12 33 22 PM" src="https://github.com/user-attachments/assets/a99f4d5f-269b-45fa-a1ac-ba2c1697014e" />

Reply:
@jonwaldstein resolved https://github.com/impress-org/givewp/pull/7900/commits/acb7926ace988390b237b43b59f8511bd3d4bfd8